lambda-buffers-compiler/src/LambdaBuffers/Compiler/TypeClass/Validate/TestCI.hs

```diff
@@ -0,0 +1,246 @@
+import LambdaBuffers.Compiler.ProtoCompat.Types qualified as P (
+  ClassDef (
+    ClassDef,
+    classArgs,
+    className,
+    documentation,
+    sourceInfo,
+    supers
+  ),
+  ClassName (ClassName),
+  CompilerInput (CompilerInput),
+  ConstrName (ConstrName),
+  Constraint (Constraint, argument, classRef, sourceInfo),
+  Constructor (Constructor),
+  ForeignRef (ForeignRef),
+  InstanceClause (..),
+  Kind (Kind),
+  KindRefType (KType),
+  KindType (KindRef),
+  LocalRef (LocalRef),
+  Module (..),
+  ModuleName (ModuleName),
+  ModuleNamePart (ModuleNamePart),
+  Product (TupleI),
+  SourceInfo (SourceInfo),
+  SourcePosition (SourcePosition),
+  Sum (Sum, constructors, sourceInfo),
+  Tuple (Tuple),
+  Ty (TyAppI, TyRefI, TyVarI),
+  TyAbs (TyAbs, sourceInfo, tyArgs, tyBody),
+  TyApp (TyApp, sourceInfo, tyArgs, tyFunc),
+  TyArg (TyArg),
+  TyBody (OpaqueI, SumI),
+  TyDef (TyDef, sourceInfo, tyAbs, tyName),
+  TyName (TyName),
+  TyRef (ForeignI, LocalI),
+  TyVar (TyVar),
+  VarName (VarName),
+ )
+
+testCompilerInput :: P.CompilerInput
+testCompilerInput = P.CompilerInput [prelude, foo2]
+
+si :: P.SourceInfo
+si = P.SourceInfo "" (P.SourcePosition 0 0) (P.SourcePosition 0 0)
+
+foo :: P.Module
+foo =
+  P.Module
+    { moduleName = P.ModuleName [P.ModuleNamePart "Foo" si] si
+    , typeDefs = [fooDef]
+    , classDefs = []
+    , instances = [cFoo]
+    , imports = [P.ModuleName [P.ModuleNamePart "Prelude" si] si]
+    , sourceInfo = si
+    }
+  where
+    fooDef =
+      P.TyDef
+        { tyName = P.TyName "Foo" si
+        , tyAbs =
+            P.TyAbs
+              { tyArgs = []
+              , tyBody =
+                  P.SumI $
+                    P.Sum
+                      { constructors =
+                          NE.singleton $
+                            P.Constructor
+                              (P.ConstrName "Foo" si)
+                              ( P.TupleI
+                                  ( P.Tuple
+                                      [ P.TyAppI $
+                                          P.TyApp
+                                            { tyFunc = P.TyRefI $ P.ForeignI $ P.ForeignRef (P.TyName "Maybe" si) (P.ModuleName [P.ModuleNamePart "Prelude" si] si) si
+                                            , tyArgs =
+                                                NE.fromList
+                                                  [ P.TyRefI $ P.ForeignI $ P.ForeignRef (P.TyName "Int" si) (P.ModuleName [P.ModuleNamePart "Prelude" si] si) si
+                                                  ]
+                                            , sourceInfo = si
+                                            }
+                                      ]
+                                      si
+                                  )
+                              )
+                      , sourceInfo = si
+                      }
+              , sourceInfo = si
+              }
+        , sourceInfo = si
+        }
```

Comment:
This is unreadable, formulate an utility module that abstracts such constructions into a more readable form.